src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -45,23 +45,26 @@ class NodeComparingVisitor : public ov::AttributeVisitor {
     void on_adapter(const std::string& name, ValueAccessor<std::shared_ptr<ov::Model>>& adapter) override {
         OPENVINO_THROW_NOT_IMPLEMENTED("Can not compare models");
     };
-    ov::AnyMap get_attributes_map() const {
+
+    ov::AnyMap visit_attributes(const std::shared_ptr<Node>& node) {
+        m_attributes_map.clear();
+        node->visit_attributes(*this);
         return m_attributes_map;
-    };
+    }
 
 private:
     ov::AnyMap m_attributes_map;
 };
 
 bool inputs_from_same_source_or_equal_constants(const std::shared_ptr<Node>& lhs, const std::shared_ptr<Node>& rhs) {
-    if (lhs->get_input_size() != rhs->get_input_size())
-        return false;
     size_t input_size = lhs->get_input_size();
+    if (input_size != rhs->get_input_size())
+        return false;
     for (size_t i = 0; i < input_size; ++i) {
-        if (lhs->input_value(i) == rhs->input_value(i))
+        if (lhs->input_value_raw(i) == rhs->input_value_raw(i))
             continue;
-        auto lhs_constant = as_type_ptr<v0::Constant>(lhs->get_input_node_shared_ptr(i));
-        auto rhs_constant = as_type_ptr<v0::Constant>(rhs->get_input_node_shared_ptr(i));
+        auto lhs_constant = as_type<v0::Constant>(lhs->get_input_node_ptr(i));
```

Comment:
If `input_value_raw` crated tuple the input node pointer should be there and could be used instead get it again.